From: Yocto Build System
Subject: [PATCH] Fix cross-compilation issues for LinuxCNC

This patch addresses various cross-compilation issues when building
LinuxCNC using Yocto/OpenEmbedded build system.

---
 src/configure.ac | 20 ++++++++++++++------
 1 file changed, 14 insertions(+), 6 deletions(-)

diff --git a/src/configure.ac b/src/configure.ac
index 1234567..abcdefg 100644
--- a/src/configure.ac
+++ b/src/configure.ac
@@ -400,9 +400,17 @@ AC_PROG_CC([$RTSCC gcc clang])
 AC_PROG_CPP()
 
 if test -z "$CC"; then
   AC_MSG_ERROR([no acceptable cc found])
-elif ! test `$CC -dumpversion | cut -d '.' -f 1` -gt 2 ; then
-  AC_MSG_ERROR([Compilers older than gcc 3.x are no longer supported])
+fi
+
+# Skip version check for cross-compilation
+if test "x$cross_compiling" != "xyes"; then
+  CC_VERSION=`$CC -dumpversion 2>/dev/null | cut -d '.' -f 1`
+  if test -n "$CC_VERSION" && test "$CC_VERSION" -le 2 2>/dev/null; then
+    AC_MSG_ERROR([Compilers older than gcc 3.x are no longer supported])
+  fi
+else
+  AC_MSG_NOTICE([Cross-compiling, skipping compiler version check])
 fi
 
 # For cross-compilation, allow skipping runtime checks
